internal/oci/notation.go

```diff
@@ -0,0 +1,225 @@
+		opt(&o)
+	}
+
+	for _, pol := range o.TrustStore.TrustPolicies {
+		certName := pol.Name
+		for _, store := range pol.TrustStores {
+			s := strings.Split(store, ":")
+			if len(s) != 2 {
+				return nil, fmt.Errorf("trust store '%s' is invalid. Trust store name must contain a store type and a store name separated by ':'. For example 'ca:fluxcd.io'", store)
+			}
+			generateTrustStore(s[0], s[1], certName, o.PublicKey)
+		}
+	}
+
+	verifier, err := verifier.New(o.TrustStore, truststore.NewX509TrustStore(dir.ConfigFS()), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &NotaryVerifier{
+		auth:     o.Keychain,
+		verifier: &verifier,
+		opts:     o.ROpt,
+		insecure: o.Insecure,
+	}, nil
+}
+
+// Verify verifies the authenticity of the given ref OCI image.
+// It returns a boolean indicating if the verification was successful.
+// It returns an error if the verification fails, nil otherwise.
+func (v *NotaryVerifier) Verify(ctx context.Context, ref name.Reference) (bool, error) {
+	url := ref.Name()
+	remoteRepo, err := oras.NewRepository(url)
+	if err != nil {
+		return false, err
+	}
+	remoteRepo.PlainHTTP = v.insecure
+
+	repo := registry.NewRepository(remoteRepo)
+
+	ss := stringResource{url}
+
+	var credentialProvider func(ctx context.Context, registry string) (oauth.Credential, error)
+
+	if v.auth != nil {
+		au, err := v.auth.Resolve(ss)
+		if err != nil {
+			return false, err
+		}
+
+		authConfig, err := au.Authorization()
+		if err != nil {
+			return false, err
+		}
+
+		credentialProvider = func(ctx context.Context, registry string) (oauth.Credential, error) {
+			if authConfig.Username != "" || authConfig.Password != "" || authConfig.IdentityToken != "" {
+				return oauth.Credential{
+					Username:     authConfig.Username,
+					Password:     authConfig.Password,
+					RefreshToken: authConfig.IdentityToken,
+				}, nil
+			}
+			return oauth.EmptyCredential, nil
+		}
+	}
+
+	repoClient := &oauth.Client{
+		Header: http.Header{
+			"User-Agent": {"flux"},
+		},
+		Cache:      oauth.NewCache(),
+		Credential: credentialProvider,
+	}
+
+	remoteRepo.Client = repoClient
+
+	i, err := remote.Image(ref, v.opts...)
+	if err != nil {
+		return false, err
+	}
+
+	d, err := i.Digest()
+	if err != nil {
+		return false, err
+	}
+
+	repoUrl := ""
+
+	lastIndex := strings.LastIndex(url, ":")
+	firstPart := url[:lastIndex]
+
+	if s := strings.Split(url, ":"); len(s) >= 2 && !strings.Contains(url, "@") {
+		repoUrl = fmt.Sprintf("%s@%s", firstPart, d)
+	}
+
+	verififyOptions := notation.VerifyOptions{
+		ArtifactReference:    repoUrl,
+		MaxSignatureAttempts: 50,
+	}
+
+	_, signatures, err := notation.Verify(ctx, *v.verifier, repo, verififyOptions)
+
+	if err != nil {
+		return false, err
+	}
+
+	if len(signatures) == 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func generateTrustStore(storeType string, storeName string, certName string, cert []byte) error {
+	// changing the path of the trust store for demo purpose.
```

Comment:
what is this comment about?

Reply:
That was an oversight on my part. I took this from the notation [example_localVerify_test.go](https://github.com/notaryproject/notation-go/blob/966c6b7e42e463522cddc673bc753f67fe52de77/example_localVerify_test.go#L165) and hadn't updated the `func`. I have since removed the `func` entirely and opted for reading the certs directly from the `[]byte` rather than writing to disk before retrieving again.